src/core/MapControls.tsx

```diff
@@ -17,17 +17,26 @@ declare global {
 
 export const MapControls = React.forwardRef<MapControlsImpl, MapControls>((props = { enableDamping: true }, ref) => {
   const { camera, ...rest } = props
-  const invalidate = useThree(({ invalidate }) => invalidate)
-  const defaultCamera = useThree(({ camera }) => camera)
-  const gl = useThree(({ gl }) => gl)
+  const { invalidate, domElement, defaultCamera } = useThree(({ invalidate, camera, gl }) => ({
```

Comment:
This is a no go, we don't have a comparison function: https://github.com/pmndrs/zustand#selecting-multiple-state-slices

Reply:
Oh, I forgot you have to add `shallow`. Thanks! I'll go back to multiple `useThree` calls.

Reply:
Should be fixed now. Should I do change anything else here?